src/transformers/utils/hub.py

```diff
@@ -499,6 +456,10 @@ def cached_file(
     return resolved_file
 
 
+# TODO: deprecate `get_file_from_repo` or document it differently?
+#       Docstring is exactly the same as `cached_repo` but behavior is slightly different. If file is missing or if
+#       there is a connection error, `cached_repo` will return None while `get_file_from_repo` will raise an error.
+#       IMO we should keep only 1 method and have a single `raise_error` argument (to be discussed).
 def get_file_from_repo(
```

Comment:
To be discussed. I tend to prefer to have a single `cached_file` method with a `raise_on_error` argument (and with the correct typing to let the user know when to expect a None value and when it's not possible). I can do it here or in a separate PR.

Reply:
Yes, I'd vote to uniformize everything under hfh's  banner.

Reply:
Actually I'd be ok for moving the logic to `huggingface_hub` but this method doesn't exist at the moment.

In hfh we have `hf_hub_download` that does the job to download/retrieve a file given a `repo_id` + `filename`. In `transformers`, `cached_file` is slightly different as it accept either a `repo_id` or a local `path` (in `path_or_repo_id` parameter), making it more versatile.

So for now I'd be more in favor of uniforming within `transformers` and then later move it to `huggingface_hub`.

Reply:
`get_file_from_repo` calls `cached_file`  (while that method has `True` as default values 
```
        _raise_exceptions_for_missing_entries=False,
        _raise_exceptions_for_connection_errors=False,
```
Not against to keep only one of them. But the changes might get larger. I would personally do this in a separate PR.

Reply:
Good idea, let's wait for a separate PR :+1: